### PR TITLE
initial control allocation support (safe to merge)

### PR DESCRIPTION
**Pull request by dagar**
Continued https://github.com/PX4/PX4-Autopilot/pull/13351 that should be safe for merging to support continued development.

Apologies for the squashed history, the rebase of https://github.com/PX4/PX4-Autopilot/pull/13351 got a little hairy and this was ultimately safer.

### CODE REVIEW

**Review by LorenzMeier — APPROVED** [review]
Makes sense to bring it in like that to keep things moving.

**Comment by LorenzMeier** [comment]
@dagar SITL tests and param metadata tests are failing.

**Comment by LorenzMeier** [comment]
Invalid unit in AVC_X_FF: N.m.s/rad

**Comment by xdwgood** [comment]
Will the corresponding document description be synchronized?

**Comment by dagar** [comment]
> Will the corresponding document description be synchronized?

I'm not sure what you're referring to, could you be more specific?

**Comment by xdwgood** [comment]
For example, how to use this new control allocation on a homemade drone?

**Comment by dagar** [comment]
> For example, how to use this new control allocation on a homemade drone?

We'll get there, but it's a bit early. At this stage I'm trying to bring in this incremental progress to enable further development.

File: src/modules/vtol_att_control/standard.cpp [hunk 1]
```diff
@@ -323,6 +323,9 @@ void Standard::update_transition_state()
 		}
 	}
 
+
+	_v_att_sp->thrust_body[0] = _pusher_throttle;
```
Comment (dagar) [inline src/modules/vtol_att_control/standard.cpp:327]: TODO:

**Comment by dagar** [comment]
The standard VTOL failures look real.

**Review by bresch — APPROVED** [review]
I went through quite carefully and did not stop anything strange.

**Comment by dagar** [comment]
The build failure was caused by some leftovers in the SITL vtol_standard aiframe.

**Pull request merged**